calculator/main.c

```diff
@@ -9,7 +9,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include "prompt.h"
+#include "compute.h"
 
 int main(int argc, const char * argv[]) {
-  return prompt();
+  return prompt(compute);
```

Comment:
In this case I pass the function name `compute` into the first parameter of `prompt`. This is possible because the compiler knows that I am referring to a function that lives in our binary. The compiler knows the location of this function and where it will be loaded into memory and makes it possible for us to pass the name of the function into another function by using a pointer. You can see where we actually execute the `char * compute(char *input)` function by looking at the code for `prompt()`

Also, notice the compiler will only allow us to do this if the function signature (return value and parameter types) all match.